mypy/messages.py

```diff
@@ -2481,6 +2507,12 @@ def match_statement_inexhaustive_match(self, typ: Type, context: Context) -> Non
             code=codes.EXHAUSTIVE_MATCH,
         )
 
+    def iteration_dependent_errors(self, iter_errors: IterationDependentErrors) -> None:
+        for error_info in iter_errors.yield_uselessness_error_infos():
+            self.fail(*error_info[:2], code=error_info[2])
+        for note_info, context in iter_errors.yield_revealed_type_infos():
+            self.reveal_type(note_info, context)
```

Comment:
It's reasonable to perform the `make_simplified_union` here, since `mypy.messages` already depends on type ops.